explorer/scan.go

```diff
@@ -54,18 +54,12 @@ func (e *Explorer) waitForSync() error {
 	return nil
 }
 
-func rhpv2Settings(ctx context.Context, publicKey types.PublicKey, netAddress string) (crhpv2.HostSettings, error) {
+func rhpv2Settings(ctx context.Context, deadline time.Time, publicKey types.PublicKey, netAddress string) (crhpv2.HostSettings, error) {
```

Comment:
instead of adding a new parameter, set a default of like 30 seconds and override with ctx.Deadline() if set.